internal/tools/modformatter/modformatter.go

```diff
@@ -0,0 +1,219 @@
+		lines = deleteElement(lines, index-removedLinesCount)
+		removedLinesCount += 1
+	}
+	return lines
+}
+
+// Format the content of a go.mod file
+func formatModFile(content string) string {
+	var requiredDeps []string
+	var replaceDeps []string
+	inReq := false
+	inRep := false
+	lines := strings.Split(content, "\n")
+	totalRemovedLines := 0
+	// Parse required and replaced dependencies
+	for idx, line := range strings.Split(content, "\n") {
+		if inReq {
+			lines = deleteElement(lines, idx-totalRemovedLines)
+			totalRemovedLines += 1
+			if strings.Contains(line, ")") {
+				inReq = false
+			} else {
+				requiredDeps = append(requiredDeps, line)
+			}
+		} else if inRep {
+			lines = deleteElement(lines, idx-totalRemovedLines)
+			totalRemovedLines += 1
+			if strings.Contains(line, ")") {
+				inRep = false
+			} else {
+				replaceDeps = append(replaceDeps, line)
+			}
+		} else if strings.HasPrefix(line, "replace") {
+			parts := strings.Split(line, " ")
+			lines = deleteElement(lines, idx-totalRemovedLines)
+			totalRemovedLines += 1
+			if parts[1][0] == '(' {
+				inRep = true
+			} else {
+				replaceDeps = append(replaceDeps, strings.Join(parts[1:], " "))
+			}
+		} else if strings.HasPrefix(line, "require") {
+			parts := strings.Split(line, " ")
+			lines = deleteElement(lines, idx-totalRemovedLines)
+			totalRemovedLines += 1
+			if parts[1][0] == '(' {
+				inReq = true
+			} else {
+				requiredDeps = append(requiredDeps, strings.Join(parts[1:], " "))
+			}
+		}
+	}
+
+	// Generate Formatted lines
+	lines = appendDeps(lines, replaceDeps, "replace", "", false)
+	lines = appendDeps(lines, requiredDeps, "require", "// indirect", true)
+	lines = appendDeps(lines, requiredDeps, "require", "// indirect", false)
+
+	// Remove leftover new lines
+	lines = removeExtraLines(lines, 1)
+
+	// join lines into string
+	return strings.Join(lines, "\n")
+}
+
+func main() {
+	flag.Parse()
+	if *path == "" {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	// Open and parse go.mod file
+	modFilePath := filepath.Join(*path, "go.mod")
+	b, err := os.ReadFile(modFilePath)
+	if err != nil {
+		fmt.Println("Error opening file %q:", err)
+		return
+	}
+	f, err := modfile.Parse("go.mod", b, func(_, v string) (string, error) {
+		return module.CanonicalVersion(v), nil
+	})
+
+	// Store every dependency already replaced in the go.mod file
+	ReplaceMap := make(map[string]bool)
+	for i := 0; i < len(f.Replace); i += 1 {
+		token := f.Replace[i].Syntax.Token
+		if len(token) < 1 {
+			continue
+		}
+		if token[0] == "replace" {
+			ReplaceMap[token[1]] = true
+		} else {
+			ReplaceMap[token[0]] = true
+		}
+
+	}
+
+	// Check if every required internal dependency is replaced
+	writeNeeded := false
+	for _, req := range f.Require {
+		if len(req.Syntax.Token) < 1 {
+			continue
+		}
+
+		modname := req.Syntax.Token[0]
+		if !strings.HasPrefix(modname, "github.com/DataDog/datadog-agent") {
+			continue
+		}
+
+		if !ReplaceMap[modname] {
```

Comment:
```suggestion
		if replace, ok := ReplaceMap[modname]; !ok {
```
I thought testing if a key is in not a map should look like that, but if your work that's great